src/services/limiter.ts

```diff
@@ -0,0 +1,82 @@
+import { JSONObject } from '@loopback/context'
+import { HttpErrors } from '@loopback/rest'
+import axios from 'axios'
+import { LimitError } from '../errors/types'
+import { blockHexToDecimal } from '../utils/block'
+
+import { WS_ONLY_METHODS } from '../utils/constants'
+
+const logger = require('../services/logger')
+
+export async function enforceEVMLimits(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  parsedRawData: Record<string, any>,
+  blockchainID: string,
+  logLimitBlocks: number,
+  altruists: JSONObject
+): Promise<void | Error> {
+  if (WS_ONLY_METHODS.includes(parsedRawData.method)) {
+    return new HttpErrors.BadRequest(
+      `We cannot serve ${parsedRawData.method} method over HTTPS. At the moment, we do not support WebSockets.`
+    )
+  } else if (parsedRawData.method === 'eth_getLogs') {
+    let toBlock: number
+    let fromBlock: number
+    let isToBlockHex = false
+    let isFromBlockHex = false
+    const altruistUrl = String(altruists[blockchainID])
+    const [{ fromBlock: fromBlockParam, toBlock: toBlockParam }] = parsedRawData.params as [
+      { fromBlock: string; toBlock: string }
+    ]
+
+    if (toBlockParam !== undefined && toBlockParam !== 'latest') {
+      toBlock = blockHexToDecimal(toBlockParam)
+      isToBlockHex = true
+    }
+    if (fromBlockParam !== undefined && fromBlockParam !== 'latest') {
+      fromBlock = blockHexToDecimal(fromBlockParam)
+      isFromBlockHex = true
+    }
+
+    if ((toBlock !== 0 || fromBlock !== 0) && altruistUrl !== 'undefined') {
+      // Altruist
+      // TODO: use a generic getHeightFromAltruist function to fetch altruist block height
+      const rawData = JSON.stringify({ jsonrpc: '2.0', id: 1, method: 'eth_blockNumber', params: [] })
+
+      let axiosConfig = {}
```

Comment:
Let's declare this object inside the `try` and put it as `const`. Objects are mutable even if they're constants—but will let us avoid changing the _type_ of the object, which is what we want.